### PR TITLE
[ALLUXIO-2542]  Fix failing mount

### PR DESCRIPTION
**Pull request by apc999**
https://alluxio.atlassian.net/browse/ALLUXIO-2542

### CODE REVIEW

**Comment by apc999** [comment]
alluxio-bot, check this please

**Comment by aaudiber** [comment]
Is this linked to the right ticket?

File: shell/src/main/java/alluxio/cli/GetConf.java [hunk 1]
```diff
@@ -14,22 +14,49 @@
...
 /**
  * Utility for printing Alluxio configuration.
  */
 public final class GetConf {
-  private static final String USAGE = "USAGE: GetConf [KEY]\n\n"
+  private static final String USAGE = "USAGE: GetConf [-bytes] [KEY]\n\n"
       + "GetConf [KEY] prints the configured value for the given key. If the key is invalid, the "
       + "exit code will be nonzero. If the key is valid but isn't set, an empty string is printed. "
-      + "If no key is specified, all configuration is printed.";
+      + "If no key is specified, all configuration is printed. If \"bytes\" option is specified, "
```
Comment (yupeng9) [inline shell/src/main/java/alluxio/cli/GetConf.java]: Will this support MB, GB?

**Comment by apc999** [comment]
@aaudiber thanks for pointing out the wrong JIRA number, it should be ALLUXIO-2542 rather than ALLUXIO-2544

File: shell/src/main/java/alluxio/cli/GetConf.java [hunk 2]
```diff
@@ -14,22 +14,49 @@
...
+      + "a value of \"1KB\" will be converted to a value of 1024.";
```
Comment (aaudiber) [inline shell/src/main/java/alluxio/cli/GetConf.java]: `If "-bytes" is specified, data size values such as "10MB" or "10GB" will be converted to bytes, e.g. "1KB" will be converted to 1024`

File: shell/src/main/java/alluxio/cli/GetConf.java [hunk 3]
```diff
@@ -14,22 +14,49 @@
...
+
+  private static final String BYTES_OPTION_NAME = "bytes";
```
Comment (jsimsa) [inline shell/src/main/java/alluxio/cli/GetConf.java]: as discussed offline, let's change this flag to `unit` which accepts B, KB, MB, GB, TB ... and optionally introduce another flag called `quantity` which accepts `size` ... this gives us the affordance to for instance introduce `time` as another quantity in the future, with supported units ns, us, ms, s, m, h, d, ...

**Comment by apc999** [comment]
@aaudiber @jsimsa @yupeng9 , I updated the PR. PTAL

File: shell/src/main/java/alluxio/cli/GetConf.java [hunk 4]
```diff
@@ -14,22 +14,82 @@
...
+      + "and \"10MB\" will return 10240. Possible unit options include B, KB, MB, GB, TP, PB";
+
+  private static final String UNIT_OPTION_NAME = "unit";
+  private static final Option UNIT_OPTION =
+      Option.builder().required(false).longOpt(UNIT_OPTION_NAME).hasArg(true)
+          .desc("unit of the value to return.").build();
+  private static final Options OPTIONS = new Options().addOption(UNIT_OPTION);
+
+  private enum Unit {
+    B(1L), KB(1L << 10), MB(1L << 20), GB(1L << 30), TB(1L << 40), PB(1L << 50);
```
Comment (jsimsa) [inline shell/src/main/java/alluxio/cli/GetConf.java]: one value per line

File: shell/src/main/java/alluxio/cli/GetConf.java [hunk 5]
```diff
@@ -14,22 +14,82 @@
...
 public final class GetConf {
-  private static final String USAGE = "USAGE: GetConf [KEY]\n\n"
-      + "GetConf [KEY] prints the configured value for the given key. If the key is invalid, the "
+  private static final String USAGE = "USAGE: GetConf [--unit] [KEY]\n\n"
+      + "GetConf prints the configured value for the given key. If the key is invalid, the "
       + "exit code will be nonzero. If the key is valid but isn't set, an empty string is printed. "
-      + "If no key is specified, all configuration is printed.";
+      + "If no key is specified, all configuration is printed. If \"--unit\" option is specified, "
+      + "values of data size configuration will be converted to a quantity in the given unit."
+      + "E.g., with \"--unit=KB\", a configuration value of \"4096\" will return 4, "
```
Comment (jsimsa) [inline shell/src/main/java/alluxio/cli/GetConf.java]: the usage string does not have `=` ... is it optional?

File: shell/src/main/java/alluxio/cli/GetConf.java [hunk 6]
```diff
@@ -40,21 +100,42 @@ public static void main(String[] args) {
...
     }
+    return 0;
+  }
+
+  /**
+   * @param args should be size 0 or 1; if size 1, the specified configuration value is printed,
+   *        otherwise all configuration is printed
+   */
+  public static void main(String[] args) {
+    int status = getConf(args);
```
Comment (jsimsa) [inline shell/src/main/java/alluxio/cli/GetConf.java]: `System.exit(getConf(args));`

File: shell/src/main/java/alluxio/cli/GetConf.java (hunk 5 above)
Comment (apc999) [inline shell/src/main/java/alluxio/cli/GetConf.java]: both are equivalent. But I removed `=` still to keep consistent

File: shell/src/main/java/alluxio/cli/GetConf.java (hunk 4 above)
Comment (apc999) [inline shell/src/main/java/alluxio/cli/GetConf.java]: done

File: shell/src/main/java/alluxio/cli/GetConf.java (hunk 6 above)
Comment (apc999) [inline shell/src/main/java/alluxio/cli/GetConf.java]: fixed

File: bin/alluxio-mount.sh [hunk 7]
```diff
@@ -143,7 +143,8 @@ function mount_ramfs_local() {
   else
     # Assuming Linux
     if [[ "$1" == "SudoMount" ]]; then
-      sudo bash -O extglob -c "mount_ramfs_linux"
+      DECL_MOUNT_LINUX=$(declare -f mount_ramfs_linux)
+      sudo bash -c "MEM_SIZE=${MEM_SIZE};TIER_PATH=${TIER_PATH};${DECL_MOUNT_LINUX};mount_ramfs_linux"
```
Comment (aaudiber) [inline bin/alluxio-mount.sh]: what is this purpose of this modification?

File: shell/src/main/java/alluxio/cli/GetConf.java [hunk 8]
```diff
@@ -40,21 +105,41 @@ public static void main(String[] args) {
...
-        System.out.println(USAGE);
-        System.exit(1);
+        printHelp("More arguments than expected");
+        return 1;
     }
+    return 0;
+  }
+
+  /**
+   * @param args should be size 0 or 1; if size 1, the specified configuration value is printed,
```
Comment (aaudiber) [inline shell/src/main/java/alluxio/cli/GetConf.java]: update javadoc

File: bin/alluxio-mount.sh (hunk 7 above)
Comment (apc999) [inline bin/alluxio-mount.sh]: This is a "bug" fix. `sudo bash -O extglob -c "mount_ramfs_linux"` doesn't work as `mount_ramfs_linux` is not defined. That's why we need `MEM_SIZE=${MEM_SIZE};TIER_PATH=${TIER_PATH};${DECL_MOUNT_LINUX};`

Comment (aaudiber) [inline bin/alluxio-mount.sh]: I see, thanks for the fix. Did this never work?

**Pull request merged**